bluecellulab/circuit/iotools.py

```diff
@@ -40,3 +44,88 @@ def parse_outdat(path: str | Path) -> dict[CellId, np.ndarray]:
     # convert outdat's index from int to CellId
     outdat.index = [CellId("", gid) for gid in outdat.index]
     return outdat.to_dict()
+
+
+def write_compartment_report(
+    output_path: str,
+    cells: CellDict,
+    report_cfg: dict,
+    source_sets: dict,
+    source_type: str,
+):
+    """Write a SONATA-compatible compartment report to an HDF5 file."""
```

Comment:
Maybe you could write in the docstring what are the expected values for source_type. Also what is source_sets